tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -274,6 +277,86 @@ def apply_yaml_postprocessing(config: ConfigNodeDict, node):
     apply_yaml_reference(config, node)
 
 
+def flatten_gitlab_ci_configuration(yml):
+    """
+    Flatten lists due to !reference tags
+    """
+    if isinstance(yml, list):
+        res = []
+        for v in yml:
+            if isinstance(v, list):
+                res.extend(flatten_gitlab_ci_configuration(v))
+            else:
+                res.append(v)
+
+        return res
+    elif isinstance(yml, dict):
+        return {k: flatten_gitlab_ci_configuration(v) for k, v in yml.items()}
+    else:
+        return yml
+
+
+def filter_gitlab_ci_configuration(yml: dict, job: str | None = None) -> dict:
+    """
+    Filters gitlab-ci configuration jobs
+
+    - job: If provided, retrieve only this job
+    """
+
+    def filter_yaml(key, value):
+        # Print only jobs
+        if key.startswith('.'):
+            return None
+
+        if job is not None:
+            return (key, value) if key == job else None
+
+        return key, value
+
+    if job is not None:
+        assert job in yml, f"Job {job} not found in the configuration"
+
+    return {node[0]: node[1] for node in (filter_yaml(k, v) for k, v in yml.items()) if node is not None}
+
+
+def print_gitlab_ci_configuration(yml: dict, sort: bool):
+    jobs = yml.items()
+    if sort:
+        jobs = sorted(jobs)
+
+    for i, (job, content) in enumerate(jobs):
+        if i > 0:
+            print()
+        yaml.safe_dump({job: content}, sys.stdout, default_flow_style=False, sort_keys=False, indent=2)
+
+
+def get_gitlab_ci_configuration(
+    input_file: str = '.gitlab-ci.yml', return_dict: bool = True, ignore_errors: bool = False
+) -> str | dict:
+    """
+    Returns the full gitlab-ci configuration by resolving all includes and applying postprocessing (extends / !reference)
+    Uses the /lint endpoint from the gitlab api to apply postprocessing
+    """
+    # Update loader/dumper to handle !reference tag
+    yaml.SafeLoader.add_constructor(ReferenceTag.yaml_tag, ReferenceTag.from_yaml)
+    yaml.SafeDumper.add_representer(YamlReferenceTagList, ReferenceTag.to_yaml)
```

Comment:
Do we still need that? If I understand correctly, the returned yaml should no longer contains !reference with the API replacing them

Reply:
This will be removed in a future PR